### DISCUSSION TITLE
How can I get gRPC Metadata (or said Response trailers) in Kong plugin ?

### DISCUSSION THREAD

**Discussion by jluncc** [Kong/kong#13218] [Help] [answered]
what i had try:  

1. kong.response.get_header(name)
2. kong.service.response.get_header(name)
3. ngx.header[name]

both can not work, so if anyone konw how to get the gRPC response Metadata value (Response trailers value) in the Kong plugin ?

**Comment by StarlightIbuki** [comment, marked as answer]
The metadata could be located in trailers, which PDK does not support to extract for now. A workaround could be to get the raw socket, read the raw request/response stream and parse the body.

**Reply by jluncc** [reply]
thanks for your answer

i am new to lua and kong plugin develop, can u give me some example code or key words to search ?

**Reply by StarlightIbuki** [reply]
https://github.com/openresty/lua-nginx-module#ngxreqsocket
It's pretty tricky to do this manually, a starting point could be: https://github.com/brimworks/lua-http-parser

**Reply by jluncc** [reply]
i will study it first, thank you

**Reply by jenstroeger** [reply]
@StarlightIbuki any plans to add support for reading trailing metadata of gRPC response messages?
